File: YZTemplate/YZCollectionViewController.xctemplate/UIViewControllerXIBObjective-C/___FILEBASENAME___.m
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

#import "___FILEBASENAME___.h"

#pragma mark - define

@interface ___FILEBASENAMEASIDENTIFIER___ ()<UICollectionViewDataSource, UICollectionViewDelegate>

@property (nonatomic, weak) IBOutlet UICollectionView *collectionView;
@property (nonatomic, strong) NSMutableArray *dataSource;

@end

@implementation ___FILEBASENAMEASIDENTIFIER___

#pragma mark - Initial Methods

#pragma mark - Override Methods
- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self setupUIComponent];
}

/*
- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
}

- (void)viewWillDisappear:(BOOL)animated {
    [super viewWillDisappear:animated];
}

-(void)viewDidDisappear:(BOOL)animated {
    [super viewDidDisappear:animated];
}
**/

#pragma mark - Setup UI Component
- (void)setupUIComponent {
    [self setupCollectionView];
}

- (void)setupCollectionView {
    
}

#pragma mark - Public Methods

#pragma mark - View Event

#pragma mark - Target Action

#pragma mark - Private Method

#pragma mark - External Delegate

#pragma mark - UICollectionViewDataSource
- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
    return self.dataSource.count;
}

static NSString * const kCellIdentifier = @"";
- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
    UICollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:kCellIdentifier forIndexPath:indexPath];
    
    return cell;
}

#pragma mark - UICollectionViewDelegate
- (void)collectionView:(UICollectionView *)collectionView didSelectItemAtIndexPath:(NSIndexPath *)indexPath {
    
}

#pragma mark - Notification

#pragma mark - Getter Methods
- (NSMutableArray *)dataSource {
    if (!_dataSource) {
        _dataSource = [NSMutableArray array];
    }
    return _dataSource;
}

#pragma mark - Setter Methods

@end

File: YZTemplate/YZCollectionViewController.xctemplate/UIViewControllerSwift/___FILEBASENAME___.swift
//
//  ___FILENAME___
//  ___PROJECTNAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

import UIKit

class ___FILEBASENAMEASIDENTIFIER___: UIViewController {
    
    @IBOutlet internal var collectionView: UICollectionView!
    var dataSource = [AnyObject]()
    
    //MARK: - Intial Methods
    
    //MARK: - Override Methods
    override func viewDidLoad() {
        super.viewDidLoad()
        
        setupUIComponent()
    }
    
    //MARK: - Setup UI Component
    func setupUIComponent() {
        setupCollectionView()
    }
    
    func setupCollectionView() {
        
    }
    
    //Mark: - Public Methods
    
    //Mark: - View Event
    
    //Mark: - Target Action
    
    //Mark: - Private Method
    
    //Mark: - External Delegate
    
    //Mark: - Notification
    
    //Mark: - Getter Methods
    
    //Mark: - Setter Methods
}

extension ___FILEBASENAMEASIDENTIFIER___: UICollectionViewDataSource {
    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
        return dataSource.count
    }
    
    private static let kCellIdentifier = ""
    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: ___FILEBASENAMEASIDENTIFIER___.kCellIdentifier, for: indexPath)
        
        return cell
    }
}

extension ViewController: UICollectionViewDelegate {
    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
        
    }
}
